### ISSUE TITLE
toTitle helper is kinda confusing

### ISSUE THREAD

**Issue by tylersticka** [cloudfour/core-hbs-helpers#32]
⚠️  **Note:** I might just be a dummy.

But I never remember that `toTitle` _only_ strips off numeric prefixes and nothing else.

For some reason I always assume it will [do this](https://www.npmjs.com/package/title-case). Which isn't even a common need now that there are capitalize helpers.

Now that Drizzle doesn't rely on filename prefixes, it could be that we don't even _need_ this helper. ¯\\\_(ツ)_/¯

**Comment by erikjung** [comment]
I wouldn't lose any sleep over removing it.

**Comment by spaceninja** [comment]
@tylersticka agreed that's a confusing name. Sounds like the helper isn't needed anymore. Can you think of any reason we shouldn't just remove it and bump the NPM package version?

**Comment by spaceninja** [comment]
Still valid in 2026. The complaint is unchanged, and the hedge at the end of the description can now be stated as fact.

### "Now that Drizzle doesn't rely on filename prefixes…"

Drizzle is finished. Its last commit was **July 2018** — eight years ago — and the npm package hasn't been touched since 2022. The use case `toTitle` was built for (turning `01 Introduction` into `Introduction` when pattern libraries were driven by numeric filename prefixes) is definitively gone.

### The helper is unchanged

Still just a numeric-prefix stripper, exactly as described:

```js
function toTitle (name) {
  var pattern = /^[0-9]{0,4}[-_]?\s+?/;
  name = name.toString();
  return name.replace(pattern, '');
}
```

So the naming confusion is real and still there. `toTitle` reads like it does [title casing](https://www.npmjs.com/package/title-case), and it doesn't — and as the description notes, the capitalize helpers cover the thing people actually expect.

### Removing it is cheaper than it sounds

Worth factoring into the decision: this package is at **0.11.0**. Under 0.x semver, dropping a helper is a minor bump, not a breaking major release. There's no long deprecation runway to plan.

Options, roughly in order of cost:

1. **Delete it.** The originating use case no longer exists, and the name actively misleads.
2. **Rename** to something honest like `stripNumericPrefix`, keeping `toTitle` as a deprecated alias.
3. **Leave it** and document the behavior more loudly.

The test coverage is there either way — `test/toTitle.spec.js` exists, though it's the thinnest spec in the repo at a single assertion, so option 2 would want a couple more cases.

**Comment by spaceninja** [comment]
Closing this. Before removing the helper I scanned the org for consumers, and `toTitle` turns out to be more entangled than "Drizzle is finished" suggests.

### Two actively-maintained repos would break immediately

`lts-patterns` (last pushed 2026-07-02) and `cpe-patterns` (2026-08-25) both register our helpers by iterating a **hardcoded list of names** and dynamically importing each one:

```js
const helpersToImport = ['all', 'and', /* … */ 'toTitle'];

helpersToImport.forEach(async (helperName) => {
  const { default: helper } = await import(`@cloudfour/hbs-helpers/lib/${helperName}.js`);
  handlebars.registerHelper(helperName, helper);
});
```

Deleting `lib/toTitle.js` makes that import throw `ERR_MODULE_NOT_FOUND` at build time. Note this is unrelated to whether the helper is *used* — the failure happens during registration, so it would take down the whole pattern library, not just any page rendering `{{toTitle}}`.

Worth flagging separately: that pattern means every filename in `lib/` is load-bearing at runtime in those repos, not merely an export key. Their comment explains why — *"We should be able to import all of the helpers from the library in a single import, but Node is getting tripped up by the package's old import structure."*

### Two dormant repos actually render it

`ceasefire-oregon` and `digital-insight` both depend on this package and call `{{toTitle …}}` in their templates. Neither has been pushed since 2016 and neither is archived, so nothing would break today — but they're the only places the helper's actual behavior is in use.

Neither `lts-patterns` nor `cpe-patterns` renders `{{toTitle}}` anywhere; they register it and never call it.

### What removal would actually cost

Not much, but more than one PR: drop `'toTitle'` from the list in both pattern libraries first, then remove the helper here and release. The sequencing matters — releasing first breaks two active builds.

That's a coordinated three-repo change to delete a helper nobody is rendering, in a package that's in maintenance mode. Not worth it right now. The naming complaint in the original description stands and is still valid; if it comes up again, this comment has the removal path.
